flink-runtime/src/main/java/org/apache/flink/runtime/io/network/buffer/NetworkBufferPool.java

```diff
@@ -181,6 +196,19 @@ public void recycle(MemorySegment segment) {
 				if (segment != null) {
 					segments.add(segment);
 				}
+
+				if ((System.nanoTime() - startTimeInNanos) / 1_000_000 >= requestSegmentsTimeoutInMillis &&
```

Comment:
The `poll` wait time is in `SECONDS` unit, I think here nanos unit is not necessary. Just in millis?

Reply:
The nanos unit is due to that I thought `System.nanoTime()` should have less overhead than `System.currentTimeInMills()`, Therefore, it might be better to use `System.nanoTime()`.